lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -356,10 +356,10 @@ def to_permission(rule):
             pair = {}
             if rule.target[0]:
                 pair['UserId'] = rule.target[0]
-            # groupid/groupname are mutually exclusive
-            if rule.target[1] and not rule.target[2]:
```

Comment:
This line wasn't right, since the index 1 and 2 will both be true since the group_id matched the foreign account ID regex - group_id will be something like `group_id: 123456789012/sg-0176895cee456bb3b/test-shertel-group`, not `group_id: 123456789012/sg-0176895cee456bb3b` or `group_id: 123456789012/test-shertel-group`.